raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -393,12 +400,12 @@ public void transitionToUnattached(int epoch) {
     }
 
     /**
-     * Grant a vote to a candidate and become a follower for this epoch. We will remain in this
+     * Grant a vote to a candidate. We will transition/remain in Unattached
      * state until either the election timeout expires or a leader is elected. In particular,
      * we do not begin fetching until the election has concluded and
      * {@link #transitionToFollower(int, int, Endpoints)} is invoked.
      */
-    public void transitionToVoted(
+    public void addVotedState(
```

Comment:
Maybe `transitionToUnattachedVoted` is better since you are doing a transition to the unattached state.